### DISCUSSION TITLE
Get Binary Mask as Inference Output

### DISCUSSION THREAD

**Discussion by d3tk** [open-edge-platform/anomalib#1191] [Q&A] [answered]
Hello, I am using the torch inference and the inference outputs the heatmap superimposed on the original image and then a segmentation. Is there anyway to get just the binary mask as the output?

**Comment by blaz-r** [comment]
Hello. I don't quite understand your question. You can get binary mask from object returned by inferencer. It's a property called `pred_mask`.

**Reply by d3tk** [reply]
I am using anomalib\tools\inference\torch_inference.py from the command line. I set the task to segmentation. Is there another way you'd recommend doing inference?

**Reply by blaz-r** [reply, marked as answer]
You will need to modify the code then, as I believe that default there is only simple visualization (the one where heat map is superimposed) and full (where you get the grid).
You will need to add `output = predictions.pred_mask` and avoid using visualizer.

**Reply by d3tk** [reply]
Thank you that fixed it. changed line 84 in torch_inferencer.py to be ouput=predictions.pred_mask then I just save it with cv2
